src/include/binder/bind_node_visitor.h

```diff
@@ -114,6 +117,32 @@ class BindNodeVisitor : public SqlNodeVisitor {
                                common::ErrorCode::ERRCODE_FEATURE_NOT_SUPPORTED);
     }
   }
+
+  std::vector<common::ManagedPointer<parser::AbstractExpression>> UnifyOrderByExpression(
+      common::ManagedPointer<parser::OrderByDescription> order_by_description,
+      const std::vector<common::ManagedPointer<parser::AbstractExpression>> &select_items) {
+    auto exprs = order_by_description->GetOrderByExpressions();
+    auto size = order_by_description->GetOrderByExpressionsSize();
+    for (size_t idx = 0; idx < size; idx++) {
+      if (exprs[idx].Get()->GetExpressionType() == terrier::parser::ExpressionType::VALUE_CONSTANT) {
+        auto constant_value_expression = exprs[idx].CastManagedPointerTo<parser::ConstantValueExpression>();
+        type::TypeId type = constant_value_expression->GetReturnValueType();
+        int64_t column_id = 0;
+        if (type == type::TypeId::INTEGER) {
+          column_id = constant_value_expression->GetInteger().val_;
+        } else if (type == type::TypeId::DECIMAL) {
+          column_id = constant_value_expression->GetReal().val_;
+        }
+        if (column_id > 0 && static_cast<size_t>(column_id) <= select_items.size()) {
+          exprs[idx] = select_items[column_id - 1];
+        } else {
+          exprs[idx] = common::ManagedPointer<parser::AbstractExpression>(new parser::ColumnValueExpression(
+              catalog::INVALID_TABLE_OID, catalog::INVALID_COLUMN_OID, type::TypeId::INVALID));
```

Comment:
Is this how you are specifying that there was an error in the binding? This seems hacky.

@mbutrovich What would be a better approach? Should it throw an exception?

Reply:
Yeah, I'd do a bounds check first rather than an if-else structure.
```
if (column_id < 1 || column_id >select_items.size()) THROW
exprs[idx] = select_items[column_id - 1];
```

Reply:
@mbutrovich THX. I've modified this according to your suggestion.

Reply:
@apavlo I‘ve committed the modification.